src/workloads/execution/ExpressiveQueries.yml

```diff
@@ -0,0 +1,207 @@
+          nextRenewalDt: {^RandomDate: {min: "2017-01-01", max: "2017-12-31"}},
+          year: {^RandomInt: {min: 1970, max: 2017}},
+          value: {^RandomInt: {min: 500, max: 10000000}},
+          address: {
+            number: {^RandomInt: {min: 1, max: 9999}},
+            street: {^RandomString: {length: {^RandomInt: {min: 30, max: 60}}}},
+            city: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}},
+            state: {^Choose: {from: ["AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "DC", "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"]}},
+            zip: {^RandomString: {length: 6}},
+            location: {position: {type: "Point", coordinates: [
+              {^RandomDouble: {min: -118.668469, max: -82.062023}},
+              {^RandomDouble: {min: 32.924436, max: 41.679986}}
+            ]}}
+          }
+        },
+        {
+          policyType: "life",
+          policyNum: {^RandomString: {length: 30}},
+          nextRenewalDt: {^RandomDate: {min: "2017-01-01", max: "2017-12-31"}},
+          insured_person: {
+            firstname: {^RandomString: {length: {^RandomInt: {min: 3, max: 10}}}},
+            lastname: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}},
+            yob: {^RandomDate: {min: "1930-01-01", max: "2016-12-31"}},
+            smoking: {^RandomInt: {min: 0, max: 1}}
+          }
+        }
+      ]}}}}
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: ManageDatabase
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        fsync: 1
+  - *Nop
+  - *Nop
+  - Repeat: 5
+    # Duration: 60 seconds
+    Database: *DB
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        createIndexes: Collection0
+        indexes:
+        - key:
+            address.state: 1
+            policies.policyType: 1
+            policies.insured_person.smoking: 1
+            gender: 1
+            dob: 1
+          name: usefulIndex
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        fsync: 1
+  - *Nop
+  - Repeat: 1
+    Database: *DB
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        drop: Collection0
+
+- Name: RunQueries
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 10
+    Database: *DB
+    Operations:
+    - OperationMetricsName: QueryNoProjectionNoIndex
+      OperationName: RunCommand
+      OperationCommand:
+        find: Collection0
+        filter: {
+          gender: "Female",
+          dob: {
+            $gte: "1990-01-01",
```

Comment:
This is treated by parser as a string, not a date. Does yaml workload description support `ISODate("...")` format?

Reply:
@jimoleary any thoughts on how we need to specify date in the filter?

Reply:
Maybe '$gte:: { "$date" : { "$numberLong" : "631152000000" } }' or  '$gte: { "$date" : "1990-01-01" }'. See [bson date](https://docs.mongodb.com/manual/reference/mongodb-extended-json/#mongodb-bsontype-Date)

The following _{^RandomDate: {min: "1990-01-01", max: "1990-01-01T00:00:00.001"}}_ will work but it's not very obvious.

If the mongodb bsontype doesn't work we should raise a ticket for it.

Reply:
[PERF-2084](https://jira.mongodb.org/browse/PERF-2084)  was created for a new generator. We should maybe consider scheduling it.

For now generating a random point date ( _{^RandomDate: {min: "1990-01-01", max: "1990-01-01T00:00:00.001"}}_ ) is the only way forward that I can see.

Supporting extended json would be a better fix but that is a large piece of work (if it is even possible at the moment).